### ISSUE TITLE
[Due for payment 2026-01-30] [$125] Keyboard Navigation: Many Pages: The links are focused and announced twice

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#76922]
## Action Performed:
Prerequisite:
- User is logged in

1. Using Chrome, open the URL: https://new.expensify.com/
2. Tab to focus on the required workspace and press Enter to activate it
3. Once the Workspace menu is displayed, tab to focus on the "Rules" menu item and press Enter to activate it
4. Tab to focus on the links "for most USD credit transactions" and "more features"
5. Note how the links are focused.

Other Occurrences:
1. On Workspaces - Expensify Card - Add bank account: the link 'Your data is secure

## Expected Result:
Each link should be focused only once.

## Actual Result:
The links are focused and announced twice:
- 'for most USD credit transactions'
- 'more features'

HTML:
<div aria-label="[https://community.expensify.com/discussion/5542/deep-dive-what-are-ereceipts/"](https://community.expensify.com/discussion/5542/deep-dive-what-are-ereceipts/%22) role="link" data-tag="pressable" tabindex="0" class="css-g5y9jx r-1otgn73 r-7q8q6z r-1471scf" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx" style="font-size: 13px;"><a href="[https://community.expensify.com/discussion/5542/deep-dive-what-are-ereceipts/"](https://community.expensify.com/discussion/5542/deep-dive-what-are-ereceipts/%22) rel="noreferrer noopener" target="_blank" role="link" class="css-1jxf684" data-testid="a" style="color: rgb(90, 176, 255); font-size: 13px; text-align: left; font-family: &quot;Expensify Neue&quot;; font-style: normal; font-weight: 400; line-height: 16px; direction: ltr; text-decoration-line: none; text-decoration-color: rgb(90, 176, 255); text-underline-position: under; text-decoration-skip-ink: none; display: inline-flex; user-select: text; cursor: pointer;">for most USD credit transactions</a></div>

<div aria-label="[https://new.expensify.com//workspaces/936055FE28804F86/more-features"](https://new.expensify.com//workspaces/936055FE28804F86/more-features%22) role="link" data-tag="pressable" tabindex="0" class="css-g5y9jx r-1otgn73 r-7q8q6z r-1471scf" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx" style="font-size: 13px;"><a href="[https://new.expensify.com//workspaces/936055FE28804F86/more-features"](https://new.expensify.com//workspaces/936055FE28804F86/more-features%22) rel="noreferrer noopener" target="_self" role="link" class="css-1jxf684" data-testid="a" style="color: rgb(1, 100, 191); font-size: 13px; text-align: left; font-family: &quot;Expensify Neue&quot;; font-style: normal; font-weight: 400; line-height: 16px; direction: ltr; text-decoration-line: none; text-decoration-color: rgb(1, 100, 191); text-underline-position: under; text-decoration-skip-ink: none; display: inline-flex; user-select: text; cursor: pointer;">more features</a></div>

## Area issue was found in:
Workspace Rules

## Failed WCAG checkpoints
WCAG 2.4.4 Link Purpose (In Context)
https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html

## User impact:
People with motion impairment can skip links that they are not interested in, avoiding the keystrokes needed to visit the referenced content and then returning to the current content. People with cognitive limitations will not become disoriented by multiple means of navigation to and from content they are not interested in. People with visual disabilities will be able to determine the purpose of a link by exploring the link's context.

## Suggested resolution:
Remove role="link" and tabindex="0" from the parent <div> since the nested <a> element already serves as the interactive link.
Ensure that only the <a> element is focusable and has the link role, and provide an aria-label to correctly match its visible label and accurately describe the control’s actual purpose.
https://gitnation.com/contents/nested-interactive-elements-an-nightmare-in-accessibility
https://adrianroselli.com/2016/12/be-wary-of-nesting-roles.html

## Environment(s)
Windows 11 Home - Chrome

## Screenshots/Videos
<img width="1918" height="972" alt="Image" src="https://github.com/user-attachments/assets/fab1dfac-fb30-4c8c-a674-81200f11fdcc" />

https://github.com/user-attachments/assets/22eee2a9-0111-477b-af17-b4045616d5ac

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** http://platform.applause.com/company/12102/products/36605/community-issues/6996814
**Issue reported by:** NA
**Slack conversation:** NA

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022009685520566693396</li>
        <li>Upwork Job ID: 2009685520566693396</li>
        <li>Last Price Increase: 2026-01-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The links are focused and announced twice:

### What is the root cause of that problem?
The links are focused twice because the wrap view of the link also has a focusable role here

https://github.com/Expensify/App/blob/fe49a6a1c23bf5bf4cf206b7574e012eb948a730/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.tsx#L60-L98
### What changes do you think we should make in order to solve the problem?
We can add `tabIndex={-1}` for the wrap pressable here, since the link text is always focusable

https://github.com/Expensify/App/blob/fe49a6a1c23bf5bf4cf206b7574e012eb948a730/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.tsx#L60-L98

### What alternative solutions did you explore? (Optional)


**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by bernhardoj** [comment]
@nkdengineer's proposal [LGTM](https://github.com/Expensify/App/issues/76922#issuecomment-3738862379)!

**Comment by nkdengineer** [comment]
@bernhardoj please help to add C+ reviewed comment.

**Comment by bernhardoj** [comment]
Oops, I forgot it.

🎀 👀 🎀 C+ reviewed

https://github.com/Expensify/App/issues/76922#issuecomment-3742627116
